drivers/io-expander/ltc4332/ltc4332.c

```diff
@@ -0,0 +1,148 @@
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+
+#include "ltc4332.h"
+#include "no_os_error.h"
+#include <stdlib.h>
+#include <string.h>
+
+/******************************************************************************/
+/**************************** Types Definitions *******************************/
+/******************************************************************************/
+
+/**
+ * @brief LTC4332 specific SPI platform ops structure
+ */
+const struct no_os_spi_platform_ops ltc4332_spi_platform_ops = {
+	.init = ltc4332_spi_init,
+	.remove = ltc4332_spi_remove,
+	.write_and_read = ltc4332_spi_write_and_read
+};
+
+/**
+ * @brief Initialize the SPI ltc4332 layer.
+ * @param desc - The SPI descriptor.
+ * @param param - The structure that contains the SPI parameters.
+ * @return 0 in case of success, -1 otherwise.
+ */
+int32_t ltc4332_spi_init(struct no_os_spi_desc **desc,
+			 const struct no_os_spi_init_param *param)
+{
+	int32_t ret;
+	struct no_os_spi_init_param temp_param;
+
+	temp_param.device_id = param->device_id;
+	temp_param.max_speed_hz = param->max_speed_hz;
+	temp_param.chip_select = param->chip_select;
+	temp_param.mode = param->mode;
+	temp_param.bit_order = param->bit_order;
+	temp_param.extra = param->extra;
+	temp_param.parent = param->parent;
+
+	if (param->parent)
+		temp_param.platform_ops = param->parent->platform_ops;
+	else
+		temp_param.platform_ops = param->platform_ops;
+
+	ret = no_os_spi_init(desc, &temp_param);
+	if (param->parent)
+		(*desc)->platform_ops = param->platform_ops;
+
+	return ret;
+}
+
+/**
+ * @brief Free the resources allocated by ltc4332_spi_init().
+ * @param desc - The SPI descriptor.
+ * @return 0 in case of success, -1 otherwise.
+ */
+int32_t ltc4332_spi_remove(struct no_os_spi_desc *desc)
+{
+	if (!desc)
+		return -EINVAL;
+
+	free(desc->extra);
+	free(desc);
+	return 0;
+}
+
+/**
+ * @brief Write and read data to/from SPI ltc4332 layer.
+ * @param desc - The SPI descriptor.
+ * @param data - The buffer with the transmitted/received data.
+ * @param bytes_number - Number of bytes to write/read.
+ * @return 0 in case of success, -1 otherwise.S
+ */
+int32_t ltc4332_spi_write_and_read(struct no_os_spi_desc *desc, uint8_t *data,
+				   uint16_t bytes_number)
+{
+	int32_t ret;
+	uint8_t *buff;
+	struct no_os_spi_init_param temp_param;
```

Comment:
delete `temp_param` and simply save the `platform_ops` - no need for the whole `no_os_spi_init_param`